src/model/PickleModel.ts

```diff
@@ -331,6 +332,18 @@ export class PickleModel {
     return undefined;
   }
 
+  loadSwapData() {
+    for (let i = 0; i < this.allAssets.length; i++) {
+      let swapProtocol = SWAP_PROTOCOLS.find((x) => {
+        return x.protocol == this.allAssets[i].protocol && x.chain == this.allAssets[i].chain;
+      });
+
+      if (swapProtocol) {
+        this.allAssets[i].swapProtocol = swapProtocol;
```

Comment:
So... basically setting the full swapProtocol structure on the pickle asset / jar / whatever... is a bit repetitive and adds unnecessary lines to pfcore which needs to be sent to every user every 3 minutes. Why not just make `swapProtocol` here be a string and having a value of `chain:protocol` or something?  

There will be multiple jars sharing swap protocols obviously, and you will be adding 4 lines that are identical to each and every jar, which is just wasteful. Then you could add a top-level element similar to `chains` where we can have `swapProtocols` with each 4-line structure included only once?

Reply:
Okay will update this.